src/Raven.Server/Documents/Indexes/Static/StaticIndexBase.VectorSearch.cs

```diff
@@ -0,0 +1,445 @@
+        }
+
+        object HandleStream(Stream stream)
+        {
+            var len = (int)stream.Length;
+            var memScope = allocator.Allocate((int)stream.Length, out Memory<byte> mem);
+            stream.ReadExactly(mem.Span);
+            return GenerateEmbeddings.FromArray(allocator, memScope, mem, vectorOptions, len);
+        }
+
+        bool IsBase64(object val) => CanTransformIntoString(val);
+
+        bool TryGetFirstNonNullElement(BlittableJsonReaderArray data, out object firstNonNull)
+        {
+            firstNonNull = data[0];
+
+            var i = 0;
+            while (IsNullValue(firstNonNull) && i < data.Length)
+                firstNonNull = data[i++];
+
+            return IsNullValue(firstNonNull) == false;
+        }
+    }
+
+    private static object VectorFromText(IndexField indexField, object value)
+    {
+        var allocator = CurrentIndexingScope.Current.IndexContext.Allocator;
+
+        if (CanTransformIntoString(value))
+            return CreateVectorValue(value);
+
+        PortableExceptions.ThrowIfNot<ArgumentException>(value is IEnumerable,
+            $"Tried to convert text into embeddings but got type {value?.GetType().FullName} which is not supported.");
+
+        var vectorList = new List<VectorValue>();
+        foreach (var item in ((IEnumerable)value)!)
+        {
+            vectorList.Add(CreateVectorValue(item));
+        }
+
+        return vectorList;
+
+        VectorValue CreateVectorValue(object valueToProcess)
+        {
+            if (IsNullValue(valueToProcess))
+                return VectorValue.Null;
+
+            var str = valueToProcess switch
+            {
+                LazyStringValue lsv => lsv,
+                LazyCompressedStringValue lcsv => lcsv,
+                string s => s,
+                LazyJsString ljs => ljs.ToString(),
+                _ => throw new NotSupportedException("Only strings are supported, but got: " + valueToProcess.GetType().FullName)
+            };
+
+            return GenerateEmbeddings.FromText(allocator, indexField.Vector, str);
+        }
+    }
+
+    /// <summary>
+    /// Determines if the given value is explicitly a string-like type.
+    /// </summary>
+    private static bool IsExplicitString(object value) => value
+        is LazyStringValue
+        or LazyCompressedStringValue
+        or string
+        or LazyJsString
+        or JsString;
+
+    /// <summary>
+    /// Determines if the given item can be converted into a string-like type.
+    /// </summary>
+    private static bool CanTransformIntoString(object item) => IsExplicitString(item)
+                                                               || IsNullValue(item);
+
+    /// <summary>
+    /// Determines if the given value is explicitly a null type
+    /// </summary>
+    private static bool IsNullValue(object value)
+    {
+        return value is null or DynamicNullObject or DynamicJsNull or JsNull;
```

Comment:
These methods were extracted from multiple locations to provide common logic across all overloads.